pkg/storage/memory/memory.go

```diff
@@ -509,9 +510,17 @@ func (s *MemoryBackend) ReadStartingWithUser(
 				targetUser = tupleUtils.GetObjectRelationAsString(userFilter)
 			}
 
-			if targetUser == t.User {
-				matches = append(matches, t)
+			if targetUser != t.User {
+				continue
 			}
+
+			if len(filter.ObjectIDs) > 0 {
+				if !slices.Contains(filter.ObjectIDs, t.ObjectID) {
+					continue
+				}
+			}
```

Comment:
Should this be moved outside of the `for` loop? There is nothing about the user filter in this block

Reply:
Totally right, fixing.